### ISSUE TITLE
Change the way in which a user "fails" a map

### ISSUE THREAD

**Issue by loldcv** [ppy/osu#6626]
A few years ago I was introduced to another rhythm game popular in Japan called "Sound Voltex" and eversince having played it and its clones myself I've always felt that the way you "crash" or "fail" a song in that game makes a lot more sense than the way osu! does it.

Please forgive me if I make a mistake explaining it here, but the way I understand it there are two modes of "health" you can have in that game. One is called "Excessive Gauge" (?) which is sort of similar to the way osu! does it. When your health goes to zero, you crash the song and are booted out of the map.

The default mode however "Effective Gauge" (?) works differently:
You start with 0% health and by playing the map correctly you "gain health" and by messing up you "lose health". There is a threshhold of for example 70% and if you can not get your health bar to be over 70% by the time the song finishes, you fail the map. This way, the user gets to play the whole song even if he fails the map in the end.

After having played osu! for years as well as having tried out this system, I feel that the system SDVX uses is a lot more enjoyable and is possibly even beneficial for player growth. When you're trying to push your limits, or playing a song over and over again trying to learn one pattern, it's frustrating to be booted out of the entire song and having to play the first 30 seconds of a 5 minute song over and over again instead of just trying again on your next run and getting more practice playing the map entirely on your first attempt.

I realize that this might be hard to implement, and can't really gauge how realistic it even is, it's just something that has been on my mind for a long time that I felt I needed to at least start a discussion about.

**Comment by bdach** [comment]
So basically taiko's HP mechanics, but in standard? I suppose this could work as a mod, but I imagine it's too drastic a change to become the *de facto* norm.

If you want to practice a map without having to worry about HP too much/at all, there's always Easy/No-Fail.

**Comment by loldcv** [comment]
Yeah, pretty much.

I understand that No Fail and Easy are mods in the game, but they don't find the bill for this. It's not that I want to take away the ability to fail or make the map easier. I just want to move the "crashing" or "failing" of a song to the end of the map, giving the player the ability to play the whole map before they fail.

Let's say I have a beatmap which is tricky, but within my skill range so that I could full combo it. It might have 2-3 parts that cause me to fail instantly, but I also have the ability to FC it. What do I do? If I put no fail and end up lucking out I have an FC with No Fail enabled. If I don't do it, I might end up getting frustrated with failing on the tricky parts before I get that one attempt that does make the full combo.

I just think it's generally a bad idea to make players fail on the spot if they "fail" the map, and the HP mechanics I proposed (which are pretty much the ones used in taiko) seemed like an obviously better solution to me.

**Comment by ThatsNiceGuy** [comment]
So this would pretty much be identical to the mechanics in Arcaea and (as mentioned previously by bdach) taiko?
If so, I really like this as the player can achieve as sense of accomplishment if they pass a really hard song by getting 70 or more on the “Effective Gauge” or as it’s known in Arcaea, “recollection rate”.
tldr I like this idea if it’s what I think it is

**Comment by zoonish** [comment]
I support, the gauge ends too quickly, and therefore I lose on the map on which I used to get rank B.
[This video shows that I did not do a single miss.](https://www.youtube.com/watch?v=3DGmuxWWEVg)
[And then this betmap.](https://osu.ppy.sh/beatmapsets/85930#mania/243830)

**Comment by ThatsNiceGuy** [comment]
Right now I think the HP drain is a bit more agressive than in osu!stable so looking into that may also help

**Comment by loldcv** [comment]
Sorry for necroing my own issue, but I was just talking about this with some friends and we thought that maybe a good compromise would be making solo play work like multiplayer. 

If you fail during solo play, your score would simply not be eligible for submission anymore. Just to make it a bit nicer, maybe they could be kept in the local rankings with an F grade so that the user can still see his past attempts.

This sounds far easier to implement to me, and could even work as a toggle in the options. Users who want to continue playing when they fail could do so, and those who do not could just leave it unticked and would fail normally.

I don't think this solution is as elegant as my initial idea, but it does take care of my main gripe with the current system in that you are no longer forced out of the map when you fail it.

**Comment by peppy** [comment]
Already have plans to do something like this, with a bit of a twist. Was actually planning in my head just last night.

**Comment by loldcv** [comment]
That's great to hear! Can't wait to see what you and the rest of the contributors have come up with. Care to share any smaller details, or give a rough outline of what is planned exactly? Just to feed my curiosity.

**Comment by peppy** [comment]
No, else I would have.

**Comment by hafizdkren** [comment]
What if than make it de-facto, you can put in onto 2 mode for player choose it. For example, I new to osu! and still using mouse. But there is a option on first day like training and competitive.

When I click training, it would be using "Effective Gauge". So I can playing from 5-7 without feeling bad.
And if I click competitive, it would be using "Excessive Gauge". So I can feel more challenging than training mode.

The other way is to make it onto "Perfect". New player will feel okay when it just get F because they fail on many note, but with better understanding over time. They being challenged to do "Full Perfect - Million Master". This technique is very successful for both new player and old player. Because there is no game over, but just feel wanna being better overtime.

So, new player can try onto this 3 gameplay mode. Normal osu! "Excessive Gauge" mode, SDVX "Effective Gauge" mode, and CyTus "No Fail" mode.

Player can choose this on Front page or on settings.
